symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/MessageService.java

```diff
@@ -1,64 +1,301 @@
+    return callAndCatchApiException(() ->
+        messageSuppressionApi.v1AdminMessagesuppressionIdSuppressPost(messageId, authSession.getSessionToken()));
+  }
+
+  /**
+   * Get the status of a particular message, i.e the list of users who the message was sent to, delivered to
+   * and the list of users who read the message.
+   *
+   * @param messageId the ID of the message to be checked
+   * @return a {@link MessageStatus} instance
+   * @see <a href="https://developers.symphony.com/restapi/reference#message-status">Message Status</a>
+   */
+  public MessageStatus getMessageStatus(@Nonnull String messageId) {
+    return callAndCatchApiException(() -> messageApi.v1MessageMidStatusGet(messageId, authSession.getSessionToken()));
+  }
+
+  /**
+   * Retrieves a list of supported file extensions for attachments.
+   *
+   * @return a list of String containing all allowed file extensions for attachments
+   * @see <a href="https://developers.symphony.com/restapi/reference#attachment-types">Attachment Types</a>
+   */
+  public List<String> getAttachmentTypes() {
+    return callAndCatchApiException(() -> podApi.v1FilesAllowedTypesGet(authSession.getSessionToken()));
+  }
+
+  /**
+   * Retrieves the details of a message given its message ID.
+   *
+   * @param messageId the ID of the message to be retrieved
+   * @return a {@link V4Message} containing the message's details, null if the message was not found
+   * @see <a href="https://developers.symphony.com/restapi/reference#get-message-v1">Get Message v1</a>
+   */
+  public V4Message getMessage(@Nonnull String messageId) {
+    return callAndCatchApiException(() -> messagesApi.v1MessageIdGet(authSession.getSessionToken(),
+        authSession.getKeyManagerToken(), messageId));
+  }
+
+  /**
+   * List attachments in a particular stream.
+   *
+   * @param streamId the stream ID where to look for the attachments
+   * @param since timestamp of the first required attachment.
+   *              This is an optional Long value representing milliseconds since January 1, 1970.
+   * @param to timestamp of the last required attachment.
+   *           This is an optional Long value representing milliseconds since January 1, 1970.
+   * @param limit maximum number of attachments to return. This optional value defaults to 50 and should be between 0 and 100.
+   * @param isSortAscending optional sort order, being ascending is set to true, otherwise set to descending.
+   *                        The default value is ascending.
+   * @return the list of attachments in the stream.
+   * @see <a href="https://developers.symphony.com/restapi/reference#list-attachments">List Attachments</a>
+   */
+  public List<StreamAttachmentItem> listAttachments(@Nonnull String streamId, Long since, Long to, Integer limit,
+      Boolean isSortAscending) {
+    final String sortDir = isSortAscending == null || isSortAscending.booleanValue() ? "ASC" : "DESC";
```

Comment:
To be replaced with an enum.

Reply:
We don't have a way to paginate for this API? for instance if there are 1000 attachments between since=0, to=10, I can only get all of them by using limit=1000?